app/coffee/Features/Announcements/AnnouncementsHandler.coffee

```diff
@@ -5,20 +5,44 @@ _ = require("lodash")
 logger = require("logger-sharelatex")
 settings = require("settings-sharelatex")
 
-module.exports =
 
-	getUnreadAnnouncements : (user_id, callback = (err, announcements)->)->
+
+
+module.exports = AnnouncementsHandler = 
+
+
+
+	_domainSpecificAnnouncements : (email)->
+		domainSpecific = _.filter settings?.domainAnnouncmentsToShow, (domainAnnouncment)->
+			matches = _.filter domainAnnouncment.domains, (domain)->
+				return email.indexOf(domain) != -1
+			return matches.length > 0
+		return domainSpecific or []
+
+
+	getUnreadAnnouncements : (user, callback = (err, announcements)->)->
 		async.parallel {
 			lastEvent: (cb)->
-				AnalyticsManager.getLastOccurance user_id, "announcement-alert-dismissed", cb
+				AnalyticsManager.getLastOccurance user?._id, "announcement-alert-dismissed", cb
 			announcements: (cb)->
 				BlogHandler.getLatestAnnouncements cb
 		}, (err, results)->
 			if err?
-				logger.err err:err, user_id:user_id, "error getting unread announcements"
+				logger.err err:err, user_id:user?._id, "error getting unread announcements"
 				return callback(err)
 			
-			announcements = _.sortBy(results.announcements, "date").reverse()
+			domainSpecific = AnnouncementsHandler._domainSpecificAnnouncements(user?.email)
```

Comment:
Worth adding some sort of error check that domain announcements have an id? If they don't they could end up impossible to dismiss? Or somewhat to enforce the schema through the settings file? Not essential, just thinking of ways this could get screwed up by accident in future.